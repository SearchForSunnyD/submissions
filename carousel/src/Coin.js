import { useState } from "react";
import heads from "./heads.png";
import tails from "./tails.png";
import "./Coin.css"

function Coin() {
  const [headsCounter, setHeadsCounter] = useState(0);
  const [tailsCounter, setTailsCounter] = useState(0);
  const [coinFace, setCoinFace] = useState(0);

  let img = coinFace === 0 ? heads : tails;

  //return 0 or 1
  function coinFlip() {
    return Math.floor(Math.random() * 2);
  }

  //flip and set the coin
  function setCoin() {

    const flip = coinFlip()

    if (flip === 0) {
      setCoinFace(0);
      setHeadsCounter(headsCounter + 1);
    } else {
      setCoinFace(1);
      setTailsCounter(tailsCounter + 1);
    }
    console.log(flip)
  }

  return (
    <div className="Coin">
      {headsCounter + tailsCounter > 0 && (
        <img className="Coin-img" src={img} alt="coin"></img>
      )}
      <div className="Counters">
        <span>heads: {headsCounter}</span> <span>tails: {tailsCounter}</span>
      </div>
      <button className="flipCoin" onClick={setCoin}> Flip </button>
    </div>
  );
}

export default Coin;
