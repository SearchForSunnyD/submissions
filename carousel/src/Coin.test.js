import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Coin from "./Coin";

it("reders without crashing", () => {
  render(<Coin />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<Coin />);
  expect(asFragment()).toMatchSnapshot();
});

it("has no image before flip", function () {
  const { container } = render(<Coin />);
  expect(container.querySelector('img[alt="coin"]')).not.toBeInTheDocument();
})

it("has an image after flip", function () {
  const { container } = render(<Coin />);
  const flip = container.querySelector(".flipCoin")
  fireEvent.click(flip)

  expect(container.querySelector('img[alt="coin"]')).toBeInTheDocument();
});
