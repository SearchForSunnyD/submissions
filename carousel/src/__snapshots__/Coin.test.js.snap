// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Coin"
  >
    <div
      class="Counters"
    >
      <span>
        heads: 0
      </span>
       
      <span>
        tails: 0
      </span>
    </div>
    <button
      class="flipCoin"
    >
       Flip 
    </button>
  </div>
</DocumentFragment>
`;
